### DISCUSSION TITLE
why i can't use a non-standard "NODE_ENV" value?

### DISCUSSION THREAD

**Discussion by pangyongsheng-tal** [vercel/next.js#64249] [Help] [answered]
### Summary

why i can't use a non-standard "NODE_ENV" value, what if If I have additional environments that need to be configured through npm commands ?  u just provide dev、test and prod,  but in practice，we often need extra env e.g. gray, rc... wtf a shit desigin it is

**Comment by icyJoseph** [comment, marked as answer]
You define your own deploy environment. 

The problem is way worse that just Next.js, many packages, actually do rely on this variable, among various things to ship debugging code, monitoring code, less optimized code, but more readable code, and enable dead code elimination

- https://overreacted.io/how-does-the-development-mode-work/

So yeah, the buck doesn't stop here, nor has it started here.

**Comment by pangyongsheng-tal** [comment]
i find a way to cope this problem,  create many env file , copy  file to .env.local in building time, e.i. 
`
"build:test": "cp .env.test .env.local && next build",
`
